kbase-extension/static/kbase/js/common/jobCommChannel.js

```diff
@@ -178,76 +181,99 @@ define([
+                request_type: requestTranslation[msgType],
+            };
 
-                let msg = {
-                    target_name: COMM_NAME,
-                    request_type: msgType,
-                };
-
-                const translations = {
-                    // backend uses `job_id` instead of `batch_id`
-                    batchId: 'job_id',
-                    jobId: 'job_id',
-                    jobIdList: 'job_id_list',
-                    pingId: 'ping_id',
-                };
-
-                for (const [key, value] of Object.entries(message)) {
-                    if (key !== 'options') {
-                        const msgKey = translations[key] || key;
-                        msg[msgKey] = value;
-                    }
-                    // convert to the batch form of the request
-                    if (key === 'batchId' && BATCH_JOB_REQUESTS[msgType]) {
-                        msg.request_type = BATCH_JOB_REQUESTS[msgType];
-                    }
-                }
+            const translations = {
+                // backend uses `job_id` instead of `batch_id`
+                batchId: 'job_id',
+                jobId: 'job_id',
+                jobIdList: 'job_id_list',
+                pingId: 'ping_id',
+            };
 
-                if (message.options) {
-                    msg = Object.assign({}, msg, message.options);
+            for (const [key, value] of Object.entries(msgData)) {
+                if (key !== 'options') {
+                    const msgKey = translations[key] || key;
+                    transformedMsg[msgKey] = value;
                 }
+                // convert to the batch form of the request
+                if (key === 'batchId' && BATCH_JOB_REQUESTS[transformedMsg.request_type]) {
+                    transformedMsg.request_type = BATCH_JOB_REQUESTS[transformedMsg.request_type];
+                }
+            }
+
+            if (msgData.options) {
+                transformedMsg = Object.assign({}, transformedMsg, msgData.options);
+            }
+
+            return transformedMsg;
+        }
 
-                this.comm.send(msg);
-                this.debug(`sending comm message: ${COMM_NAME} ${msgType}`);
-                resolve();
+        /**
+         * Sends the messages in this.messageQueue to the JobManager in the kernel.
+         * If there's no comm channel ready, the message will be added to the
+         * message queue.
+         *
+         * If no arguments are supplied to sendCommMessage, the stored messages (if any)
+         * will be sent.
+         *
+         * @param {string} msgType - message type; will be one of the
+         *                 keys in the requestTranslation object (optional)
+         *
+         * @param {object} msgData - additional parameters for the request,
+         *                 such as jobId or jobIdList, or an 'options' object (optional)
+         */
+        sendCommMessage(msgType, msgData) {
```

Comment:
This method signature is useful for testing, so I left it in; `_sendCommMessages` takes messages from the `messageQueue` and sends them to the backend.